crates/blockchain/mempool.rs

```diff
@@ -45,8 +49,13 @@ impl MempoolInner {
     /// Remove a transaction from the pool with the transaction pool lock already taken
     fn remove_transaction_with_lock(&mut self, hash: &H256) -> Result<(), StoreError> {
         if let Some(tx) = self.transaction_pool.get(hash) {
-            if matches!(tx.tx_type(), TxType::EIP4844) {
-                self.blobs_bundle_pool.remove(hash);
+            if matches!(tx.tx_type(), TxType::EIP4844)
+                && let Some(h) = self.blobs_bundle_pool.remove(hash)
+            {
+                for commitment in &h.commitments {
+                    self.blobs_bundle_by_versioned_hash
+                        .remove(&kzg_commitment_to_versioned_hash(commitment));
```

Comment:
Can two transactions share versioned hashes? If that's the case, we'll need to keep a reference counter here.